--- a/cmake/helpers/CheckDependentLibraries.cmake
+++ b/cmake/helpers/CheckDependentLibraries.cmake
@@ -221,5 +221,9 @@ gdal_check_package(Shapelib "Enable Shapelib support (not recommended, internal
                    DISABLED_BY_DEFAULT)
-if (NOT GDAL_USE_SHAPELIB)
+if (DEFINED OGR_ENABLE_DRIVER_SHAPE AND NOT OGR_ENABLE_DRIVER_SHAPE)
+  set(GDAL_USE_SHAPELIB_INTERNAL
+      OFF
+      CACHE BOOL "Set ON to build shape driver with internal shapelib" FORCE)
+elseif (NOT GDAL_USE_SHAPELIB)
   set(GDAL_USE_SHAPELIB_INTERNAL
       ON
       CACHE BOOL "Set ON to build shape driver with internal shapelib" FORCE)
--- a/gdal.cmake
+++ b/gdal.cmake
@@ -320,7 +320,7 @@ function (is_sharp_embed_available res)
     if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.21 AND
         ((CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0) OR
-         (CMAKE_C_COMPILER_ID STREQUAL "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 19.0)))
+         (CMAKE_C_COMPILER_ID MATCHES "^(Apple)?Clang$" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 19.0)))
         # CMAKE_C_STANDARD=23 only supported since CMake 3.21
         set(TEST_SHARP_EMBED
           "static const unsigned char embedded[] = {\n#embed __FILE__\n};\nint main() { (void)embedded; return 0;}"
